Description: Fix the Autoconf compiler probe for C23
Origin: vendor, https://src.fedoraproject.org/rpms/gmp/blob/48233878dabed24130a8868e3eed48e394315805/f/gmp-6.3.0-c23.patch
Forwarded: not-needed
Last-Update: 2026-09-16

Imported unchanged from Fedora packaging commit 48233878dabed24130a8868e3eed48e394315805.
SHA256 of original patch: 44fb7f1b37db0994535a3a0364988411312af26f9fb071ced275f69f2e5c4be1

commit 14837bacbbd80804a11fee2016f660d132bf8aec
Author: Marc Glisse <marc.glisse@inria.fr>
Date:   Wed Jan 29 22:38:02 2025 +0100

    Complete function prototype in acinclude.m4 for C23 compatibility

diff --git a/acinclude.m4 b/acinclude.m4
index fddb5fb07..4fca12de2 100644
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -609,7 +609,7 @@ GMP_PROG_CC_WORKS_PART([$1], [long long reliability test 1],
 
 #if defined (__GNUC__) && ! defined (__cplusplus)
 typedef unsigned long long t1;typedef t1*t2;
-void g(){}
+void g(int,t1 const*,t1,t2,t1 const*,int){}
 void h(){}
 static __inline__ t1 e(t2 rp,t2 up,int n,t1 v0)
 {t1 c,x,r;int i;if(v0){c=1;for(i=1;i<n;i++){x=up[i];r=x+1;rp[i]=r;}}return c;}
